src/transformers/onnx/config.py

```diff
@@ -345,11 +369,31 @@ def generate_dummy_inputs(
             batch_size = compute_effective_axis_dimension(batch_size, fixed_dimension=OnnxConfig.default_fixed_batch)
             dummy_input = self._generate_dummy_images(batch_size, num_channels, image_height, image_width)
             return dict(preprocessor(images=dummy_input, return_tensors=framework))
+        elif (
+            isinstance(preprocessor, FeatureExtractionMixin) and preprocessor.model_input_names[0] == "input_features"
+        ):
+            # If dynamic axis (-1) we forward with a fixed dimension of 2 samples to avoid optimizations made by ONNX
+            batch_size = compute_effective_axis_dimension(batch_size, fixed_dimension=OnnxConfig.default_fixed_batch)
+            dummy_input = self._generate_dummy_audio(batch_size, sampling_rate, time_duration, frequency)
+            return dict(preprocessor(dummy_input, return_tensors=framework))
         else:
             raise ValueError(
                 "Unable to generate dummy inputs for the model. Please provide a tokenizer or a preprocessor."
             )
 
+    def generate_dummy_inputs_onnxruntime(self, reference_model_inputs: Mapping[str, Any]) -> Mapping[str, Any]:
```

Comment:
If I'm not mistaken, this function doesn't do anything - why do we need it?

Reply:
Hi @lewtun , this is true that this function does not do anything for existing models. However, this function can be overridden in some cases where the model inputs and ONNX Runtime inputs are different. This was needed when exporting the decoder in encoder-decoder models using encoder_outputs. For example: Check the following Decoder config in optimum where I am using the function [DecoderOnnxConfig](https://github.com/huggingface/optimum/pull/420/files#diff-c27ea812737bc6ccfe34f92a4ff0d1ec473a41b8c8012bfdb08bb22a46104ddeR342). 

Since we are waiting for the optimum PR to merge and migrating the changes this is no longer needed in the current merge.

But we still need to update the `VisionEncoderDecoderConfig` using encoder_outputs in that case we would need such function. Should we merge with this and utilise this in new `VisionEncoderDecoder` PR (In case the user wants to create the PR for this change, this option would become easier for him)? Or we could remove it from here and add the function along with the new PR?

Reply:
I see, thanks for the clarification! Looking at the `optimum` code, it seems like we use this function to add new fields to the ORT inputs - is there a reason we can't capture that logic in a single `generate_dummy_inputs()` function associated with the ONNX config?

(I'm happy to include this function as is, just trying to understand if we really need it or not)

Reply:
The function (at least in this case) update the existing keys in the input dict, but the values remains the same.

For exporting these models we would require 2 different input sets. This is because the model.forward has a different input signature (since this is the full model before export) that the generated ONNX model (only decoder is exported). Therefore we need a way to alter the existing model inputs to run inference with both models in validate_model_outputs.

However, I think creating a separate function is a much cleaner way. But I am open to any suggestions. We can add the logic in the `generate_dummy_inputs` by adding a new argument to the function maybe called `ort_inputs=True/False` and for these models we return different sets of inputs. But this will require updation of all the OnnxConfigs having this function.

Reply:
OK great, now I understand well why we need this - thanks. IMO it's fine to include this function in this PR if we add a small note in the docstring like "Override to run inference with seq2seq models which have the encoder and decoder exported as separate ONNX files."

Reply:
Done